### PR TITLE
Delete todo

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: assets/js/index.js [hunk 1]
```diff
@@ -80,12 +91,18 @@ function formatDate(date) {
...
     <p class="todo"  title="${formatDate(createdAt)}">${text}</p>
     <i class="fa-solid fa-pen-to-square edite-icon"></i>
     <i class="fa-solid fa-xmark remove-icon"></i>
     </li>`).join("");
+  const removeIcons = Array.from(document.getElementsByClassName("remove-icon"));
+  removeIcons.forEach(removeIcon => {
+    removeIcon.addEventListener("click", function (event) {
+      deleteTodo(event.target.parentNode.id);
+    })
+  });
```
Comment (blacksrc) [inline assets/js/index.js]: Extract this to another function and call it `bindDeleteEvent`.

File: assets/js/index.js [hunk 2]
```diff
@@ -80,12 +91,18 @@ function formatDate(date) {
 function showTasks(tasks) {
   todoList.innerHTML = tasks.sort(
     (a, b) => new Date(b.createdAt) -  new Date(a.createdAt)
-  ).map(({text, createdAt}) => `<li class="todo-item">
+  ).map(({ text, createdAt, id: taskId }) => `<li id="${taskId}" class="todo-item">
```
Comment (blacksrc) [inline assets/js/index.js:90]: Where are we using this `id`?

File: assets/js/index.js [hunk 3]
```diff
@@ -108,4 +125,12 @@ function updateFilterCounts(tasks) {
 
 function showNetworkError() {
   backendErrorMessage.innerText = "Something went wrong!";
-}
\ No newline at end of file
+}
+
+async function showTasksFirstRender() {
```
Comment (blacksrc) [inline assets/js/index.js]: Call this function `renderTasks`.

File: assets/js/index.js (hunk 1 above)
Comment (hedita) [reply assets/js/index.js]: Done.

File: assets/js/index.js (hunk 2 above)
Comment (hedita) [reply assets/js/index.js:90]: For deleting task.

File: assets/js/index.js (hunk 3 above)
Comment (hedita) [reply assets/js/index.js]: Done.

File: assets/js/index.js (hunk 2 above)
Comment (blacksrc) [reply assets/js/index.js:90]: How? Would you elaborate a bit more?

Comment (hedita) [reply assets/js/index.js:90]: Sure.I will send a voice message.

File: assets/js/index.js [hunk 4]
```diff
@@ -108,4 +120,21 @@ function updateFilterCounts(tasks) {
...
 function showNetworkError() {
   backendErrorMessage.innerText = "Something went wrong!";
-}
\ No newline at end of file
+}
+
+async function renderTasks() {
+  const tasks = await getTasks();
+  showTasks(tasks);
+  updateFilterCounts(tasks);
```
Comment (blacksrc) [inline assets/js/index.js:124]: If you look closely into the code this part of code is repeated in two other places. Maybe you could use this function instead?

File: assets/js/index.js [hunk 5]
```diff
@@ -80,12 +91,13 @@ function formatDate(date) {
 function showTasks(tasks) {
   todoList.innerHTML = tasks.sort(
     (a, b) => new Date(b.createdAt) -  new Date(a.createdAt)
-  ).map(({text, createdAt}) => `<li class="todo-item">
+  ).map(({ text, createdAt, id: taskId }) => `<li id="${taskId}" class="todo-item">
     <input type="checkbox" />
     <p class="todo"  title="${formatDate(createdAt)}">${text}</p>
```
Comment (blacksrc) [inline assets/js/index.js:92]:
```suggestion
    <p class="todo" title="${formatDate(createdAt)}">${text}</p>
```

File: assets/js/index.js (hunk 4 above)
Comment (hedita) [reply assets/js/index.js:124]: Done.

Comment (blacksrc) [reply assets/js/index.js:124]: It's not done yet!

File: assets/js/index.js [hunk 6]
```diff
@@ -23,29 +23,35 @@ async function postTodo(todoText) {
 
     if (response.status !== 201) {
       errorMessage.innerText = error;
+    }
```
Comment (blacksrc) [inline assets/js/index.js]: indentation ;)

Comment (hedita) [reply assets/js/index.js]: Fixed.

**Pull request merged**